### ISSUE TITLE
Unstated dependency version?

### ISSUE THREAD

**Issue by thisisnic** [msperlin/yfR#11]
I installed the package and tried to run the examples in `yf-vignette.Rmd` but got some errors:

```
> library(yfR)
> # set options for algorithm
> my_ticker <- 'FB'
> first_date <- Sys.Date() - 30
> last_date <- Sys.Date()
> # fetch data
> df_yf <- yf_get(tickers = my_ticker, 
+                      first_date = first_date,
+                      last_date = last_date)

── Running yfR for 1 stocks | 2022-05-05 --> 2022-06-04 (30 days) ──

ℹ Downloading data for benchmark ticker ^GSPC
ℹ (1/1) Fetching data for FB
! 	- not cached
✖ 	- error in download..
ℹ Binding price data
Error in `vec_slice()`:
! `x` must be a vector, not NULL.
Run `rlang::last_error()` to see where the error occurred.
Warning messages:
1: Unknown or uninitialised column: `ticker`. 
2: Unknown or uninitialised column: `threshold_decision`. 
3: Unknown or uninitialised column: `ticker`. 
4: Unknown or uninitialised column: `price_adjusted`. 
> rlang::last_error()
<error/vctrs_error_scalar_type>
Error in `vec_slice()`:
! `x` must be a vector, not NULL.
---
Backtrace:
 1. yfR::yf_get(tickers = my_ticker, first_date = first_date, last_date = last_date)
 2. yfR:::calc_ret(...)
 3. dplyr::lag(P)
 5. vctrs::vec_slice(inputs$x, seq_len(xlen - n))
Run `rlang::last_trace()` to see the full context.
> rlang::last_trace()
<error/vctrs_error_scalar_type>
Error in `vec_slice()`:
! `x` must be a vector, not NULL.
---
Backtrace:
    ▆
 1. ├─yfR::yf_get(tickers = my_ticker, first_date = first_date, last_date = last_date)
 2. │ └─yfR:::calc_ret(...)
 3. │   └─dplyr::lag(P)
 4. │     ├─vctrs::vec_c(...)
 5. │     └─vctrs::vec_slice(inputs$x, seq_len(xlen - n))
 6. └─vctrs:::stop_scalar_type(`<fn>`(NULL), "x", `<env>`)
 7.   └─vctrs:::stop_vctrs(...)
 8.     └─rlang::abort(message, class = c(class, "vctrs_error"), ..., call = vctrs_error_call(call))
> session_info()
Error in session_info() : could not find function "session_info"
> sessionInfo()
R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C               LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8     LC_MONETARY=en_GB.UTF-8   
 [6] LC_MESSAGES=en_GB.UTF-8    LC_PAPER=en_GB.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] yfR_0.0.3

loaded via a namespace (and not attached):
 [1] pillar_1.7.0     compiler_4.1.2   tools_4.1.2      xts_0.12.1       digest_0.6.29    evaluate_0.14    lifecycle_1.0.1  tibble_3.1.6    
 [9] lattice_0.20-45  pkgconfig_2.0.3  rlang_1.0.2      cli_3.3.0        DBI_1.1.1        rstudioapi_0.13  curl_4.3.1       yaml_2.2.2      
[17] xfun_0.29        fastmap_1.1.0    dplyr_1.0.9      stringr_1.4.0    knitr_1.37       generics_0.1.2   vctrs_0.4.1      grid_4.1.2      
[25] tidyselect_1.1.1 glue_1.6.1       R6_2.5.0.9000    fansi_0.5.0      rmarkdown_2.11   purrr_0.3.4      TTR_0.24.3       magrittr_2.0.1  
[33] ellipsis_0.3.2   htmltools_0.5.2  assertthat_0.2.1 quantmod_0.4.18  utf8_1.2.2       stringi_1.7.6    crayon_1.4.1     zoo_1.8-9 
```

This may be due to package versioning issues; I hadn't updated any packages in a long time and after updating all of my packages, I successfully ran the examples.  Here's my `sessionInfo()` from after my update:

```
> sessionInfo()
R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C               LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8     LC_MONETARY=en_GB.UTF-8   
 [6] LC_MESSAGES=en_GB.UTF-8    LC_PAPER=en_GB.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] yfR_0.0.3

loaded via a namespace (and not attached):
 [1] pillar_1.7.0     compiler_4.1.2   tools_4.1.2      xts_0.12.1       lifecycle_1.0.1  tibble_3.1.7     lattice_0.20-45  pkgconfig_2.0.3 
 [9] rlang_1.0.2      DBI_1.1.2        cli_3.3.0        rstudioapi_0.13  curl_4.3.2       dplyr_1.0.9      stringr_1.4.0    generics_0.1.2  
[17] vctrs_0.4.1      hms_1.1.1        grid_4.1.2       tidyselect_1.1.2 glue_1.6.2       R6_2.5.1         fansi_1.0.3      purrr_0.3.4     
[25] readr_2.1.2      tzdb_0.3.0       TTR_0.24.3       magrittr_2.0.3   scales_1.2.0     ellipsis_0.3.2   assertthat_0.2.1 quantmod_0.4.20 
[33] colorspace_2.0-3 utf8_1.2.2       stringi_1.7.6    munsell_0.5.0    crayon_1.5.1     zoo_1.8-10 
```

So there may be a package in there which you need to add a minimum version for in your DESCRIPTION.

**Comment by msperlin** [comment]
Hi @thisisnic, I tested here (Linux Mint/ubuntu) and its working fine. Also, github-actions also tested in Windows/MAc and it is also working there.

Perhaps you reached the YF call limit? If so, it releases the block in a couple of minutes. Can you try it again ?

**Comment by msperlin** [comment]
btw, did  you upgrade your packages in the last month? there was a change in quantmod which affected yfR.

**Comment by thisisnic** [comment]
Hi @msperlin, I tried multiple times on separate dates but without any previous calls to the API, and it failed both times with the same error.  Unfortunately, I can't try it again, as it only failed before I upgraded my packages, so it works fine now.  The 2 different outputs above show a change in quantmod version, so I think you're right - that could well be the package which needs the version number adding in the DESCRIPTION file.
